sonar-dotnet-shared-library/src/main/java/org/sonar/plugins/dotnet/tests/BranchCoverage.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * SonarSource :: .NET :: Shared library
+ * Copyright (C) 2014-2020 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+package org.sonar.plugins.dotnet.tests;
+
+import java.util.Objects;
+
+// This class is responsible to keep SQ/SC metrics related to branch coverage
+class BranchCoverage {
+  private int line;
+  private int conditions;
+  private int coveredConditions;
+
+  public BranchCoverage(int line, int conditions, int coveredConditions){
+    this.line = line;
+    this.conditions = conditions;
+    this.coveredConditions = coveredConditions;
+  }
+
+  public int getLine() {
+    return line;
+  }
+
+  public int getConditions() {
+    return conditions;
+  }
+
+  public int getCoveredConditions() {
+    return coveredConditions;
+  }
+
+  public void add(int conditions, int coveredConditions){
+    this.conditions += conditions;
+    this.coveredConditions += coveredConditions;
+  }
+
+  @Override
+  public boolean equals(Object o) {
+    if (this == o) {
+      return true;
+    }
+
+    if (o == null || getClass() != o.getClass()) {
```

Comment:
also, would you like to add a UT for this class to ensure full code coverage? [(see sonarcloud)](https://sonarcloud.io/component_measures?id=org.sonarsource.dotnet%3Asonar-csharp&metric=new_uncovered_conditions&pullRequest=3229&selected=org.sonarsource.dotnet%3Asonar-csharp%3Asonar-dotnet-shared-library%2Fsrc%2Fmain%2Fjava%2Forg%2Fsonar%2Fplugins%2Fdotnet%2Ftests%2FBranchCoverage.java&view=list)

I think it's a low hanging fruit